src/components/form/settings.vue

```diff
@@ -44,40 +77,152 @@ except according to the terms contained in the LICENSE file.
+      <template v-else #body>
+        <p>
+          {{ $t('webFormsSetting.enketoConfirmation.description') }}
+        </p>
+      </template>
+    </confirmation>
   </div>
 </template>
 
-<script>
+<script setup>
+import { inject, computed, ref, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useRouter } from 'vue-router';
 import FormDelete from './delete.vue';
+import Confirmation from '../confirmation.vue';
 
 import useRoutes from '../../composables/routes';
 import { modalData } from '../../util/reactivity';
 import { useRequestData } from '../../request-data';
+import { apiPaths } from '../../util/request';
 
-export default {
-  name: 'FormSettings',
-  components: { FormDelete },
-  inject: ['alert'],
-  setup() {
-    const { form } = useRequestData();
-    const { projectPath } = useRoutes();
-    return { form, deleteModal: modalData(), projectPath };
-  },
-  methods: {
-    afterDelete() {
-      const message = this.$t('alert.delete', { name: this.form.nameOrId });
-      this.$router.push(this.projectPath())
-        .then(() => { this.alert.success(message); });
-    }
+defineOptions({
+  name: 'FormSettings'
+});
+
+const alert = inject('alert');
+
+const { t } = useI18n();
+const router = useRouter();
+const { form } = useRequestData();
+const { projectPath, formPath } = useRoutes();
+
+const deleteModal = modalData();
+
+const afterDelete = () => {
+  const message = t('alert.delete', { name: form.nameOrId });
+  router.push(projectPath())
+    .then(() => { alert.success(message); });
+};
+
+const webformsEnabled = ref(form.webformsEnabled);
+watch(() => form.dataExists, () => {
+  webformsEnabled.value = form.webformsEnabled;
+});
+
+const previewPath = computed(() => formPath(
+  form.projectId,
+  form.xmlFormId,
+  'preview'
+));
+
+const confirmModalState = ref(false);
+
+const confirm = computed(() => {
+  const result = {
+    state: confirmModalState.value,
+    noText: t('action.cancel'),
+    awaitingResponse: form.awaitingResponse
+  };
+  if (webformsEnabled.value) {
+    result.title = 'ODK Web Forms';
+    result.yesText = t('webFormsSetting.webformsConfirmation.useOdkWebForms');
+  } else {
+    result.title = 'Enketo';
+    result.yesText = t('webFormsSetting.enketoConfirmation.useEnketo');
   }
+  return result;
+});
+
+const showConfirmation = () => {
+  confirmModalState.value = true;
+};
+
+const hideAndReset = () => {
+  webformsEnabled.value = form.webformsEnabled;
+  confirmModalState.value = false;
+};
+
+const setWebformsEnabled = () => {
+  form.request({
+    method: 'PATCH',
+    url: apiPaths.form(form.projectId, form.xmlFormId),
+    data: { webformsEnabled: webformsEnabled.value },
+    patch: ({ data }) => {
+      form.updatedAt = data.updatedAt;
+      form.webformsEnabled = data.webformsEnabled;
+    }
+  })
+    .catch(() => {
+      webformsEnabled.value = form.webformsEnabled;
+    })
+    .finally(() => {
+      confirmModalState.value = false;
+    });
```

Comment:
I feel like if there's an error response, the modal should stay open, and we should show the alert within it. That's what we usually do for modals. What do you think about something like this?

```suggestion
    .then(hideAndReset)
    .catch(noop);
```